automerge-backend/src/encoding.rs

```diff
@@ -632,4 +636,15 @@ impl ColData {
         }
         Ok(len)
     }
+
+    pub fn deflate(&mut self) {
```

Comment:
Do you mean deflated rather than decompressed? The data is written to the column uncompressed using the `encode` method.

I have thought about doing some more type heavy stuff here. We could add a phantom type parameter to `ColData` and then only allow encoding deflated columns to the document format. This would also have the nice property of allowing us to guarantee that we don't accidentally compress columns when writing them to changes (rather than the document format) as that's not allowed. That would look a little like this:

```rust
trait DeflateState;
enum NotDeflate{}
enum MaybeDeflated{}

impl DeflateState for NotDeflated{}
impl DeflateState for MaybeDeflated{}


struct ColData<T> {
    pub col: u32,
    pub data: Vec<u8>,
    phantom: PhantomData<T>
}

impl ColData<NotDeflated> {
    fn deflate(self) -> ColData<MaybeDeflated>
}

/// later in ColumnEncoder::finish
    fn finish(self) -> (Vec<u8>, HashMap<u32, Range<usize>>) {
        let mut coldata: Vec<ColData<NotDeflated>> = vec![
            self.insert.finish(COL_INSERT),
            self.action.finish(COL_ACTION),
        ];
        ...
    }
```

I haven't gone down this road as it seems like it's adding a lot of complexity for something that we should catch in unit tests and interop tests anyway, adding a debug only field does make a lot of sense though so I've gone ahead and done that.

Reply:
Yeah, we could go down the typestate route but I agree it may add more complexity at this stage when we might not need it. Basically I think until the entire binary format reaches 1.0 and so is stable we shouldn't build in too much complexity for these things, especially if they are internal and we can change them later.